R/IDateTime.R

```diff
@@ -302,24 +302,27 @@ as.POSIXlt.ITime = function(x, ...) {
 #   lubridate routines do not return integer values.
 ###################################################################
 
+do_fastroutine = function(x)
+  (inherits(x, 'POSIXct') && identical(attr(x, 'tzone', exact=TRUE), 'UTC')) ||
+  inherits(x, 'ITime')
 second  = function(x) {
-  if (inherits(x, 'POSIXct') && identical(attr(x, 'tzone', exact=TRUE), 'UTC')) {
+  if (do_fastroutine(x)) {
     # if we know the object is in UTC, can calculate the hour much faster
     as.integer(x) %% 60L
```

Comment:
does it make sense to change this like into `as.integer(x %% 60)` to prevent overflow?
```
as.integer(5e9) %% 2L
[1] NA
Warning message:
NAs introduced by coercion to integer range 
```
or maybe not worth...
if `%%` is much faster on integer than double then I don't think it makes much sense as issues like such overflow is quite unlikely

Reply:
#3781